pkg/monitoring/metrics/virt-controller/vmistats_collector.go

```diff
@@ -202,6 +206,32 @@ func getVMIMachine(vmi *k6tv1.VirtualMachineInstance) (guestOSMachineType string
 	return
 }
 
+func getVMIPod(vmi *k6tv1.VirtualMachineInstance) string {
+	if vmi == nil {
+		return ""
+	}
+
+	objs, err := kvPodInformer.GetIndexer().ByIndex(cache.NamespaceIndex, vmi.Namespace)
+	if err != nil {
+		return ""
+	}
+
+	for _, obj := range objs {
+		pod, ok := obj.(*k8sv1.Pod)
+		if !ok {
+			continue
+		}
+
+		if pod.Labels["kubevirt.io/created-by"] == string(vmi.UID) && pod.Status.Phase == k8sv1.PodRunning {
+			if vmi.Status.NodeName == pod.Spec.NodeName {
+				return pod.Name
+			}
+		}
+	}
+
+	return ""
+}
```

Comment:
can't we use `vmi.Status.ActivePods` to reduce the number of pods checked?

Reply:
this is already reduced by node namespace.. I did it with active pods at first but @fossedihelm suggested to use node and vmi namespace

Reply:
The `ActivePods` are equivalent to the pod labeled with `"kubevirt.io/created-by": string(vmi.UID)`. So it will not reduce the number of iterations

Reply:
```
	// ActivePods is a mapping of pod UID to node name.
	// It is possible for multiple pods to be running for a single VMI during migration.
	ActivePods map[types.UID]string `json:"activePods,omitempty"`
```

Reply:
Oh now I got you!

Reply:
so you mean update to this: @machadovilaca @fossedihelm 
 
```
 func getVMIPod(vmi *k6tv1.VirtualMachineInstance) string {
	if vmi == nil || len(vmi.Status.ActivePods) == 0 {
		return ""
	}

	for podUID, nodeName := range vmi.Status.ActivePods {
		if nodeName == vmi.Status.NodeName {
			// Retrieve the pod by its UID directly from the informer
			pod, exists, err := kvPodInformer.GetIndexer().GetByKey(vmi.Namespace + "/" + string(podUID))
			if err != nil || !exists {
				continue
			}
			if pod.(*k8sv1.Pod).Status.Phase == k8sv1.PodRunning {
				return pod.(*k8sv1.Pod).Name
			}
		}
	}

	return ""
}
```

Reply:
ok so I tried this and it does not work for migrations
`func getVMIPod(vmi *k6tv1.VirtualMachineInstance) string {
	if vmi == nil || len(vmi.Status.ActivePods) == 0 {
		return ""
	}

	for podUID, nodeName := range vmi.Status.ActivePods {
		if nodeName == vmi.Status.NodeName {
			pod, exists, err := kvPodInformer.GetIndexer().GetByKey(vmi.Namespace + "/" + string(podUID))
			if err != nil || !exists {
				continue
			}
			if pod.(*k8sv1.Pod).Status.Phase == k8sv1.PodRunning {
				return pod.(*k8sv1.Pod).Name
			}
		}
	}

	return ""
}`

Reply:
`pod, exists, err := kvPodInformer.GetIndexer().GetByKey(vmi.Namespace + "/" + string(podUID))`
This will not work.
GetByKey uses the namespace/name not the namespace/uuid

Reply:
>> I don't think we can get pod from informer starting from the uuid

Seems like you are right.
I think we should leave as is